Library.Application/Services/BookCheckoutService.cs

```diff
@@ -0,0 +1,94 @@
+﻿using AutoMapper;
+using Library.Application.Dtos;
+using Library.Infrastructure.Repositories;
+using Library.Domain.Models;
+using Library.Application.Interfaces;
+
+namespace Library.Application.Services;
+
+/// <summary>
+/// Сервис, обеспечивающий CRUD-операции для работы с записями о выдаче книг читателям.
+/// </summary>
+public class BookCheckoutService(
+    BookCheckoutRepository bookCheckoutRepository,
```

Comment:
Вам необходимо в проекте `Library.Application` избавиться от зависимости `Library.Infrastructure`. Оставьте зависимость только от `Library.Domain`.
Необходимо зависеть от абстракции а не от реализации. В данном случае у вас `BookCheckoutRepository` имеет интерфейс `IRepository`, вот и указывайте его.
```suggestion
    IRepository<BookCheckout, int> bookCheckoutRepository,
```
Аналогично ниже. Таким образом в вашем проекте не должно остаться зависимости от InMemory реализации.